addons/account/models/company.py

```diff
@@ -196,6 +204,19 @@ def _check_fiscalyear_last_day(self):
             if rec.fiscalyear_last_day > max_day:
                 raise ValidationError(_("Invalid fiscal year last day"))
 
+    @api.depends('fiscal_position_ids.foreign_vat')
+    def _compute_multi_vat_foreign_country(self):
+        company_to_foreign_vat_country = {
+            val['company_id'][0]: val['country_ids']
+            for val in self.env['account.fiscal.position'].read_group(
+                domain=[('company_id', 'in', self.ids), ('foreign_vat', '!=', False)],
```

Comment:
I think there is a misunderstanding here.
Either there is something that I missed, either it's my code that isn't clear and I should improve the code clarity.

Here is what I think my code does:
From what I understand and what I test for the `read_group`, it will return a list of dict.
As I give `company_id` as the only grouping key, I won't have 2 dict for the same company.
In this dict, I will have all the `foreign_vat` fiscal_position **that concern the grouping key company**.

As this is a shitty API to use, I do what a lot of other places do: I  change it for a simple `company: foreign_vat_fiscal_positionS`.

Later on, I iterate on the company and just get the result that concerns this company.
I don't think there is any colliding issue here that would set it up for another one.

Tell me if you see something wrong in my thoughts

Reply:
You're right ; I missed the group by part :man_facepalming: